### PR TITLE
fix(useExternal): avoid wrong option

### PR DESCRIPTION
**Pull request by vaakian**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution
Avoid potential wrong option.

For example, if user specifies `type: 'js'` then `css` attribute is useless and will be omitted, this PR can help it at type level.

#### ❌ wrong
image

#### ✅ correct
image

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: packages/hooks/src/useExternal/index.ts [hunk 1]
```diff
@@ -1,10 +1,23 @@
...
-}
+};
+
+export type Options =
+  | CssOptions
+  | JsOptions
+  | {
+      type: never;
+      js?: Partial<HTMLScriptElement>;
+      css?: Partial<HTMLStyleElement>;
```
Comment (miracles1919) [inline packages/hooks/src/useExternal/index.ts]: 这里有什么用么

Comment (vaakian) [reply packages/hooks/src/useExternal/index.ts]: 这里表示当省略了type，而通过 path 的后缀自动判断类型，这个时候ts无法确定类型，所以两个选项都是允许的。

**Comment by miracles1919** [comment]
我在你的基础上稍微调整了下，看看这样怎么样

File: packages/hooks/src/useExternal/index.ts [hunk 2]
```diff
@@ -1,23 +1,24 @@
...
-    };
+type DefaultOptions = {
+  type?: 'js' | 'css';
+  js?: Partial<HTMLScriptElement>;
+  css?: Partial<HTMLStyleElement>;
+};
+
+type OptionsType = 'js' | 'css' | undefined;
+type Opt<T> = T extends 'js' ? JsOptions : T extends 'css' ? CssOptions : DefaultOptions;
+export type Options = Opt<OptionsType>;
```
Comment (vaakian) [inline packages/hooks/src/useExternal/index.ts]: 没问题。不过我感觉这里有点啰嗦了，最后的结果其实就等于下面的类型：
```ts
export type Options = JsOptions | CssOptions | DefaultOptions;
```

**Pull request merged**